packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkBridgeExt.kt

```diff
@@ -47,3 +57,72 @@ internal fun Map<*, *>.toWritableMap(): WritableNativeMap {
     }
     return map
 }
+
+/**
+ * Casts the ReadableArray to a List that strictly contains non-null objects of type HashMap.
+ * @throws TypeCastException if at least one of the list items is not of type HashMap.
+ */
+internal fun ReadableArray.toHashMapArrayList(): List<HashMap<*, *>> {
+    return this.toKotlinArrayList().toMutableList().mapNotNull {
+        if (it is HashMap<*, *> || it == null) {
+            it as HashMap<*, *>?
+        } else {
+            throw TypeCastException(
+                "Cannot convert ReadableArray to ArrayList of HashMap(s): " +
+                    "$it is not of type HashMap<*,*>",
+            )
+        }
+    }
+}
+
+/**
+ * Recursively converts the ReadableMap to a HashMap which only contains Kotlin stdlib objects,
+ * such as ArrayList, HashMap and the raw types.
+ * @throws IllegalArgumentException if the readable map is malformed (eg. it contains a HashMap
+ * or ArrayList, instead of ReadableMap and ReadableArray respectively).
+ */
+internal fun ReadableMap.toKotlinHashMap(): HashMap<String, Any> {
+    val hashMap: HashMap<String, Any> = this.toHashMap()
+    val iterator: Iterator<*> = hashMap.keys.iterator()
+
+    while (iterator.hasNext()) {
+        val key = iterator.next() as String
+        when (getType(key)) {
+            ReadableType.Null, ReadableType.Boolean, ReadableType.Number, ReadableType.String -> {}
+            ReadableType.Map -> hashMap[key] = (Assertions.assertNotNull(getMap(key)) as ReadableMap).toKotlinHashMap()
+            ReadableType.Array -> hashMap[key] = (Assertions.assertNotNull(getArray(key)) as ReadableArray).toKotlinArrayList()
+            else -> throw IllegalArgumentException("Could not convert object with key: $key.")
```

Comment:
`Assertions` should be used only in tests and we cannot allow to throw anyway. So if `getType` returns map or array, why should it be other type in the value?

```suggestion
            ReadableType.Map -> hashMap[key] = getMap(key).toMap()
            ReadableType.Array -> hashMap[key] = getArray(key).toList()
            else -> { log to logcat and telemetry (if it is available) }
```